src/ts/cart-page/usePromoCode.ts

```diff
@@ -0,0 +1,144 @@
+
+let cartProducts: CartLocalStor[] = [];
+const localStor = localStorage.getItem("cartProducts");
+if (localStor) {
+  cartProducts = JSON.parse(localStor);
+}
+
+function showPromoCodes() {
+  if (!promoCodeInput || !foundPromo) return;
+  const localStorPromoCodes = localStorage.getItem("OSpromoCodes");
+  outer: for (let i = 0; i < availablePromoCodes.length; i++) {
+    if (promoCodeInput.value.toUpperCase() === availablePromoCodes[i].id) {
+      foundPromo.style.display = "block";
+      if (!localStorPromoCodes) {
+        foundPromo.innerHTML = `${availablePromoCodes[i].name} - ${availablePromoCodes[i].disc}% <span class="applied-codes__span found-promo__btn_add">ADD</span>`;
+        break outer;
+      } else {
+        let usedPromoCodes: AllPromoCodes[] = [];
+        usedPromoCodes = JSON.parse(localStorPromoCodes);
+        for (let j = 0; j < usedPromoCodes.length; j++) {
+          if (promoCodeInput.value.toUpperCase() === usedPromoCodes[j].id) {
+            foundPromo.innerHTML = `${usedPromoCodes[j].name} - ${usedPromoCodes[j].disc}%`;
+            break outer;
+          } else {
+            foundPromo.innerHTML = `${availablePromoCodes[i].name} - ${availablePromoCodes[i].disc}% <span class="applied-codes__span found-promo__btn_add">ADD</span>`;
+            break outer;
+          }
+        }
+      }
+    } else {
+      foundPromo.style.display = "none";
+    }
+  }
+}
+
+function addPromoCode(e: Event) {
+  if ((e.target as HTMLElement).classList.contains("found-promo__btn_add")) {
+    applyPromoCode(e.target as HTMLElement);
+  }
+}
+
+function removePromoCode(e: Event) {
+  if ((e.target as HTMLElement).classList.contains("promo__btn_drop")) {
+    deletePromoCode(e.target as HTMLElement);
+  }
+}
+
+function applyPromoCode(target: HTMLElement) {
+  target.style.display = "none";
+  if (
+    !promoCodeInput ||
+    !startPrice ||
+    !resultPrice ||
+    !resultPriceMoney ||
+    !appliedCodes
+  )
+    return;
+
+  const localStorPromoCodes = localStorage.getItem("OSpromoCodes");
+  let usedPromoCodes: AllPromoCodes[] = [];
+  if (localStorPromoCodes) {
+    usedPromoCodes = JSON.parse(localStorPromoCodes);
+  }
+
+  availablePromoCodes.forEach((item) => {
+    if (promoCodeInput.value.toUpperCase() === item.id) {
+      usedPromoCodes.push(item);
+      localStorage.setItem("OSpromoCodes", JSON.stringify(usedPromoCodes));
+    }
+  });
+  createAppliedPromo();
+  changePromoCodeMoney();
+}
+
+export function createAppliedPromo() {
+  const localStorPromoCodes = localStorage.getItem("OSpromoCodes");
+  if (!appliedCodes || !localStorPromoCodes) return;
+  if (localStorPromoCodes.length === 0) return;
+  appliedCodes.style.display = "block";
+  appliedCodes.innerHTML = "";
+  const title = document.createElement("h3");
+  title.innerHTML = "Applied codes";
+  appliedCodes.appendChild(title);
+
+  let usedPromoCodes: AllPromoCodes[] = [];
+  if (localStorPromoCodes) {

```

Comment:
looks like you already checked `localStorPromoCodes`, this check is redundant

Reply:
Fixed by https://github.com/PetrAlexkulakov/Online-Store/pull/18/commits/6eeacd6535b97e0843e6fcf33cf5c42108ff0cea